model.go

```diff
@@ -387,6 +389,11 @@ func (m *model) EnvironVersion() int {
 	return m.EnvironVersion_
 }
 
+// SetEnvironVersion implements Model.
+func (m *model) SetEnvironVersion(v int) {
```

Comment:
Thanks Alvin, I know other places in this code do the "implements foo" comment but it isn't good enough anymore. Can we write some better documentation? I would suggest copying and pasting the comments made above for the interface.

Reply:
Can we please also add a unit test for this?

Reply:
Sure thing thanks for the valuable suggestions Tom :D